libs/go/libp2p_node/dht/dhtpeer/dhtpeer.go

```diff
@@ -787,6 +792,7 @@ func (dhtPeer *DHTPeer) handleNewDelegationConnection(conn net.Conn) {
 				}
 			}
 		}()
+		wg.Wait()
```

Comment:
this is functionally equivalent to not creating the goroutine, i.e., reduces the degree of parallelism to 1